src/generate.js

```diff
@@ -338,8 +338,17 @@ module.exports = {
 
         // copy local defined models to dist directory keeping the local folder structure.
         task('copy-ts-model', ['cleanup-dist', 'read-rest-api'], function () {
-           return gulp.src([path.join(source, 'ts') + '/**/*.ts'], {base: source})
-                .pipe(gulp.dest('model', {cwd: dest}));
+            // SDCISA-4685 respect application argument 'models' to copy local ts
+            if (params.models() && fs.existsSync(params.models())) {
+                log.debug('Copying ts models from ', colors.magenta(params.models()));
+                return gulp.src([params.models() + '/**/*.ts'], {cwd: source})
+                    .pipe(gulp.dest(path.join('model', 'ts'), {cwd: dest}));
+            } else {
+                const sourcePath = path.join(source, 'ts');
+                log.debug('Copying ts models from ', colors.magenta(sourcePath));
```

Comment:
I think would be nice if the static part of the two log messages would differ. Would make it easier to spot which branch effectively executed. Relying only on the variable has potential to mislead researches.

For example:
```js
log.debug( 'Copying ts models from (arg)', ..... );
// ...
log.debug( 'Copying ts models from (default)', ..... );
```
Or whatever.